srml/contracts/src/wasm/prepare.rs

```diff
@@ -74,6 +74,23 @@ impl<'a> ContractModule<'a> {
 		Ok(())
 	}
 
+	/// Ensures that tables declared in the module are not too big.
+	fn ensure_table_size_limit(&self, limit: u32) -> Result<(), &'static str> {
+		let err_msg = self.module.table_section()
+			.and_then(|table_section| table_section.entries().first())
```

Comment:
Why only first?

Reply:
In Wasm 1.0, [only one table](https://www.w3.org/TR/wasm-core-1/#tables%E2%91%A0) may be defined or imported per module. This deserves a code comment for sure.

If there could be multiple tables, we'd have to choose further restrictions, like is the total size of the tables limited or each table's size or whatever.

Reply:
Yup, I was just thinking that if we were to lift this (rather conservative) restriction, we might just forget it to patch here : )